control-plane/rest/openapi-specs/v0_api_spec.yaml

```diff
@@ -3816,6 +3922,87 @@ components:
       required:
         - cordonlabels
         - drainlabels
+    RegisterAppNode:
+      example:
+        endpoint: '10.1.0.5:50055'
+        labels:
+          openebs.io/csi-node.nvme-ana: true,
+          openebs.io/zone: us-east-1a
+      type: object
+      properties:
+        endpoint:
+          type: string
+          format: url
+        labels:
+          type: object
```

Comment:
Need to better define what these labels are, ex: are these control-plane labels, are these csi labels, k8s labels etc..
Though csi and kubernetes terms don't quite belong "here", so might need another type of abstraction, maybe a map of maps?

Reply:
maybe two different label fields, like `labels` and `control_plane_labels`. We can go with map of maps but I think it would be better to have separate fields dedicated to it's own purpose, if that makes it easier to understand.

Reply:
the different labels fields work for 2, but not for 3.
let's say you also wanted to keep k8s only labels?

Reply:
As discussed will, keep it for later.